### ISSUE TITLE
No error or warning on potentially invalid JSX syntax

### ISSUE THREAD

**Issue by hi-ogawa** [oxc-project/oxc#9770]
There are some odd code found in https://github.com/oxc-project/oxc/pull/9769, which oxc currently processes without errors or warnings.

```jsx
// 036.jsx
<a left=<a /> right=<b>monkeys /> gorillas</b> />;
// 039.jsx
<a>foo}</a>;
// 040.jsx
<a>foo></a>;
```

https://playground.oxc.rs/#eNpNj8tug0AMRX8lmnVU0qfUlLKqsm3VbrMxg6HTDjayhzQo4t87AwJl5ec9174Ya/Ymyza7+6ebHz0fKYeNxzq8xpgVG3HNd8zLomX6xUFTr2Fx3oPmWVnE+uVIk/550Rc185hnsEwedteTYpqYrWGzvxjpKQUdKMDZ7IP0uDXeUVhytdzhWgxtyX6pggBpzdLOjXFrOhBFSUDwnv8+MfRC731QV+GhJxscR7tZ3QnG3RN+gCDpikjeEyLmq8HMDCANxsMM6t3u9jG+4JQ9BKze0HoQSPhIqsFrQlmusMHpPyQoPX5xLxZb6Fa31pGr3eJnmYKwP8TTk+iEUrLG52fgOP4DtKKUTQ==

Comparing with other parsers, I'm not sure where this should be a hard error or a warning. Some examples:
- esbuild (warning) https://esbuild.github.io/try/#dAAwLjI1LjEALS1sb2FkZXI9anN4AC8vIDAzNi5qc3gKPGEgbGVmdD08YSAvPiByaWdodD08Yj5tb25rZXlzIC8+IGdvcmlsbGFzPC9iPiAvPjsKLy8gMDM5LmpzeAo8YT5mb299PC9hPjsKLy8gMDQwLmpzeAo8YT5mb28+PC9hPjs
- typescript (warning?) https://ast-explorer.dev/#eNpNjkEOgjAQRa/SzJpYEokJiL2Ae1fdFCy1WKekRSMh3N0BgmE1M+/l588IDgpo1UfFOtiuhwQ6Av3Q6T+oCXDO0uPp0MavxFIxp5v+QpMLFqx50F6Jl8enHuLMjA/WORVLXgm6zxKXfL7lReP9VHK1mSzdG7EYKvZUPEpkTIJTaN7K6JsO0XqUULA8T1a3Pnq1eJ9xJnGC6QelxkOk
- typescript eslint (error) https://ast-explorer.dev/#eNpNjcEOgjAMhl9l6RkdicZExB19BE+7DFJwODayDSMhvLsdhuip7ff1b2cwUECnXirUXg8RMhgIxGnAL9hhMNomXhPnnOWH074Lb2lLxQw28UqVC+Z1+6C+Er2zT5xCYq3z2hgVSl4Jmi/SrvnzlheNc0vJ1WaO+b8Rq6HHjh7P0jImAete3dEH7ayEgoBREUOUkP38DVUcPYa0sMaSSHdpjn7EhBZpF1g+wWxQBg==

**Comment by overlookmotel** [comment]
Fixed in #9777.
